### PR TITLE
feat: support versioned continue-as-new

### LINKED ISSUE #205: continueAsNew does not support specifying a new orchestration version
## Summary

The `continueAsNew()` API does not expose a way to specify a new orchestration version for the restarted instance. The underlying protobuf field (`CompleteOrchestrationAction.newVersion`, field 4) exists in the proto definition, but the TypeScript SDK never populates it.

## Problem

`OrchestrationContext.continueAsNew()` only accepts `(newInput: any, saveEvents: boolean)`. There is no parameter to pass a new version.

In `src/utils/pb-helper.util.ts`, `newCompleteOrchestrationAction()` does not accept a `newVersion` parameter and never calls `.setNewversion()` on the `CompleteOrchestrationAction` — the field is always left unset.

This prevents version migration scenarios for long-running/eternal orchestrations.

## What exists today

- **Proto layer**: `CompleteOrchestrationAction.newVersion` (field 4) is defined in `orchestrator_service.proto` and supported by all backends
- **TS API**: `continueAsNew(newInput: any, saveEvents: boolean)` — no version parameter
- **pb-helper**: `newCompleteOrchestrationAction()` doesn't accept or set `newVersion`

## Proposed fix

Follow the pattern from the .NET SDK ([microsoft/durabletask-dotnet#682](https://github.com/microsoft/durabletask-dotnet/pull/682)):

1. Add a `ContinueAsNewOptions` interface with `newInput`, `saveEvents`, and `newVersion` properties
2. Add a `continueAsNew(options: ContinueAsNewOptions)` overload (or add optional `newVersion` parameter)
3. In `pb-helper.util.ts`, update `newCompleteOrchestrationAction()` to accept and set `newVersion` via `.setNewversion()`
4. Wire through from `runtime-orchestration-context.ts` `setContinuedAsNew()`

## Related

- .NET SDK implementation: microsoft/durabletask-dotnet#682
- Proto field: `CompleteOrchestrationAction.newVersion` (field 4) in `orchestrator_service.proto`

### PR DESCRIPTION
**Pull request by YunchuWang**
# Summary

## What changed?
- Added an optional third `newVersion` parameter to core and compatibility `continueAsNew` APIs.
- Propagated the version through `CompleteOrchestrationAction.newVersion` and the in-memory backend's next `ExecutionStarted` event.
- Added focused protocol, in-memory restart, compatibility, documentation, and changelog coverage.

## Why is this change needed?
- Eternal orchestrations need a safe history-reset boundary for migrating to a newly registered orchestration version.
- A positional optional parameter is the smallest backwards-compatible TypeScript API: existing one- and two-argument calls keep their current behavior.

## Issues / work items
- Fixes #205

---

# Project checklist
- [ ] Release notes are not required for the next release
  - [x] Otherwise: Notes added to `CHANGELOG.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] Breaking change? No

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot CLI
- AI-assisted areas/files: API design, core/runtime/protobuf/in-memory propagation, compatibility adapter, tests, and documentation
- What you changed after AI output: The implementation was iteratively reviewed against the issue, sibling .NET behavior, repository conventions, and targeted validation results.

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Passed
  - `npm test -w @microsoft/durabletask-js -- --runTestsByPath test\orchestration_executor.spec.ts test\in-memory-backend.spec.ts`
  - `npm test -w durable-functions -- --runTestsByPath test\unit\orchestration-context.spec.ts`
  - Targeted ESLint on all changed TypeScript files
  - `npm run build -w durable-functions` (builds core and compatibility packages)
  - `git diff --check` with the repository's existing CRLF test-file convention

## Manual validation (only if runtime/behavior changed)
- Environment (OS, Node.js version, components): Windows, Node.js 24, in-memory Durable Task backend
- Steps + observed results:
  1. Continued an orchestration from the default version to `2.0.0`.
  2. Observed `CompleteOrchestrationAction.newVersion` set to `2.0.0` and the restarted orchestration context report `ctx.version === "2.0.0"`.
- Evidence (optional): Covered by focused unit and in-memory tests above.

---

# Notes for reviewers
- No generated protobuf files or shared proto definitions were changed.

### CODE REVIEW

File: packages/durabletask-js/src/testing/in-memory-backend.ts [hunk 1]
```diff
@@ -594,6 +594,7 @@ export class InMemoryOrchestrationBackend {
     if (status === pb.OrchestrationStatus.ORCHESTRATION_STATUS_CONTINUED_AS_NEW) {
       // Handle continue-as-new
       const newInput = completeAction.getResult()?.getValue();
+      const newVersion = completeAction.getNewversion()?.getValue();
       const carryoverEvents = completeAction.getCarryovereventsList();
```
[the remark this answers is not in the document]
Comment (YunchuWang) [reply packages/durabletask-js/src/testing/in-memory-backend.ts:1060]: Fixed in fd7dffe. The in-memory restart now uses the current ExecutionStarted version when newVersion is absent, while an explicitly present wrapper still overrides it. Added a three-generation regression covering 1.0.0 -> explicit 2.0.0 -> omitted -> 2.0.0.

**Review by bachuv — DISMISSED** [review]

**Review by kaibocai — APPROVED** [review]
